main.go

```diff
@@ -146,4 +146,9 @@ func main() {
 		setupLog.Error(err, "problem removing subscriptions")
 		os.Exit(1)
 	}
+
+	setupLog.Info("removing console plugin CR")
```

Comment:
Cannot removal of subscriptions be thought of as uninstallation? Only during operator deletion would that step be invoked.

Reply:
@bipuladh We are in process of removing subscription logic from here and we are adding odf-operator subscription as an owner to the owned subscriptions so whenever odf-operator subscription gets deleted other subscriptions also get deleted. you can do the same for console also.

Reply:
In this case, I don't think he can. He's right in that we don't have anything that is cluster-scoped to use as an OwnerRef for the plugin. As such, we can't rely on k8s garbage collection to remove it for us. That said, I don't think we want to remove the ConsolePlugin every time we exist the controller manager... I was doing that with Subscriptions because deleting them basically does nothing from the OLM perspective.

I think I would rather see something that arbitrarily runs when an exit signal is encountered[1]. Here I think you should check either the odf-operator CSV or the odf-operator Deployment itself to see if its marked for deletion. If so, try to delete the ConsolePlugin, retrying only if the Delete() operation does not return OK or ErrNotFound.

[1] Something like this: https://stackoverflow.com/questions/10385551/get-exit-code-go